### DISCUSSION TITLE
Static assertion failed due to requirement: "missing value callback overload for production"

### DISCUSSION THREAD

**Discussion by ashermkn1** [foonathan/lexy#114] [Q&A] [answered]
Hi,

I am currently working through Advent of Code and am trying to parse something resembling this:

U826,R410,D652,R549,D698,L805,U814,L364,U905,L96,U997,L689
L1008,D451,L146,D628,R877,U486,L464,U815,L119,U208,R686

Here is my parser I have right now:
```
typedef std::pair<char, int> step;
    namespace grammar {
        namespace dsl = lexy::dsl;
        struct instruction {
            static constexpr auto rule =
                    LEXY_LITERAL_SET(LEXY_LIT("U"), LEXY_LIT("R"), LEXY_LIT("D"), LEXY_LIT("L")) + dsl::integer<int>;
            static constexpr auto value = lexy::construct<step>;
        };
        struct wire {
            static constexpr auto rule = dsl::terminator(dsl::eol).opt_list(dsl::p<instruction>, dsl::sep(dsl::comma));
            static constexpr auto value = lexy::as_list<std::vector<step>>;
        };
        struct parser {
            static constexpr auto rule = dsl::times<2>(dsl::p<wire>);
            static constexpr auto value = lexy::construct<std::pair<std::vector<step>, std::vector<step>>>;
        };
 }
```

However, I am getting a compile time error (as the title suggests):
<img width="858" alt="image" src="https://user-images.githubusercontent.com/28579754/211617888-07b12dbd-b3cf-4720-9d24-569660ec506b.png">

I couldn't find anything about this error in the docs, so I'd appreciate any help you all could give :)

**Comment by foonathan** [comment, marked as answer]
The error indicates that you don't have a callback for `grammar::instruction` that accepts an `int&`. Your rule produces only a single `int`, but the constructor for `pair<int, int>` takes two integers.

You probably need to convert `U`, `R`, etc. into an appropriate value. This can be done via lexy::symbol_table: https://lexy.foonathan.net/reference/dsl/symbol/#symbol_table

**Reply by ashermkn1** [reply]
Awesome, symbol_table is exactly what I'm looking for! Thank you
